File: DataGridview ComboBox usage/[C#][VB]-DataGridview ComboBox usage/C# and VB/Demo2/Extensions/DatagridViewExtensions.vb
Module DatagridViewExtensions
    <System.Diagnostics.DebuggerStepThrough()> _
    <System.Runtime.CompilerServices.Extension()> _
    Public Function IsComboBoxCell(ByVal sender As DataGridViewCell) As Boolean
        Dim Result As Boolean = False
        If sender.EditType IsNot Nothing Then
            If sender.EditType Is GetType(DataGridViewComboBoxEditingControl) Then
                Result = True
            End If
        End If
        Return Result
    End Function
    <System.Diagnostics.DebuggerHidden()> _
    <System.Runtime.CompilerServices.Extension()> _
    Public Sub ExpandColumns(ByVal sender As DataGridView)
        For Each col As DataGridViewColumn In sender.Columns
            col.AutoSizeMode = DataGridViewAutoSizeColumnMode.AllCells
        Next
    End Sub
End Module

File: DataGridview ComboBox usage/[C#][VB]-DataGridview ComboBox usage/C# and VB/Demo1_C/Extensions/DatagridViewExtensions.cs
using System.Windows.Forms;

namespace Demo1_C
{
    public static class DatagridViewExtensions
    {
        public static bool IsComboBoxCell(this DataGridViewCell sender)
        {
            bool Result = false;
            if (sender.EditType != null)
            {
                if (sender.EditType == typeof(DataGridViewComboBoxEditingControl))
                {
                    Result = true;
                }
            }
            return Result;
        }

        public static void ExpandColumns(this DataGridView sender)
        {
            foreach (DataGridViewColumn col in sender.Columns)
            {
                col.AutoSizeMode = DataGridViewAutoSizeColumnMode.AllCells;
            }
        }
    }
}